### PR TITLE
Gatsby v5 migration and merge conflicts resolved

### PR DESCRIPTION
**Pull request by Aviral0702**
**Description**

This PR fixes aims to migrate Layer5 website to gatsby v5 and react-18.
Here I have worked upon #4646 and resolved the merge conflicts.

**Notes for Reviewers**

**[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: src/collections/blog/2022/2022-05-10-dockercon-22-hashicorp-talk/index.mdx [hunk 1]
```diff
@@ -115,7 +115,8 @@ MeshMap is the world's only visual designer for Kubernetes and all cloud native
 <iframe src="https://www.youtube.com/embed/3DPZafR8VWM" loading="lazy" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen ></iframe>
 </div>
 
-<p><strong>Lee Calcote and Nic Jackson packed a great deal of information in this talk. Find the recording below. The Meshery Extension is now out! Try now, and Share your Experience </strong>
+<p>
```
Comment (sudhanshutech) [inline src/collections/blog/2022/2022-05-10-dockercon-22-hashicorp-talk/index.mdx:118]: try to exclude these unnecessary line changes

File: src/collections/careers/digital-marketing-internship/index.mdx [hunk 2]
```diff
@@ -51,7 +51,7 @@ import { Col } from "../../../reusecore/Layout";
         </p>
     </div>
     <div className="qualification-section body-sections">
-        <Col xs={12} lg={6} className="qualification-col">
+        <Col $xs={12} $lg={6} className="qualification-col">
```
Comment (sudhanshutech) [inline src/collections/careers/digital-marketing-internship/index.mdx:54]: use of `$` ?

File: src/templates/blog-tag-list.js [hunk 3]
```diff
@@ -7,7 +7,7 @@ import { graphql } from "gatsby";
 
 export const query = graphql`query BlogsByTags($tag: String!) {
   allMdx(
-    sort: {fields: [frontmatter___date], order: DESC}
+    sort: {frontmatter: {date: DESC}}
```
Comment (sudhanshutech) [inline src/templates/blog-tag-list.js:10]: is this the new way of writing query?

**Comment by sudhanshutech** [comment]
@Aviral0702 is the build failing only because we need to update mdx ?
Why the `emotion/style` package has conflict dependency issue?

File: src/collections/careers/digital-marketing-internship/index.mdx (hunk 2 above)
Comment (Aviral0702) [reply src/collections/careers/digital-marketing-internship/index.mdx:54]: It indicates passing of the prop to a styled-component as transient props.

File: src/templates/blog-tag-list.js (hunk 3 above)
Comment (Aviral0702) [reply src/templates/blog-tag-list.js:10]: Yes this is new way for writing graphql queries.

File: src/collections/careers/digital-marketing-internship/index.mdx (hunk 2 above)
Comment (sudhanshutech) [reply src/collections/careers/digital-marketing-internship/index.mdx:54]: Thanks for the clarification... :)

File: src/collections/blog/2022/2022-05-10-dockercon-22-hashicorp-talk/index.mdx (hunk 1 above)
Comment (Aviral0702) [reply src/collections/blog/2022/2022-05-10-dockercon-22-hashicorp-talk/index.mdx:118]: For these type of changes I am referring to latest repo of Layer5 and then making the changes according to it. I will avoid them from now.

**Comment by l5io** [comment]
🚀 Preview for commit 5ab108eb661a8e6eaad4c162ca8f6d4d51116587 at: https://66d9f5ef21f96a5bb79da073--layer5.netlify.app

**Comment by Aviral0702** [comment]
Time taken to build the Current running version of the Layer5 website: 626.78 sec
Time taken to build website after migration with this PR : 526.63 sec.

**Comment by l5io** [comment]
🚀 Preview for commit 2c6c5409bfd118b1ab69e8f499cafd73185b6b0d at: https://66dd713a14f944ccd82d9a24--layer5.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 63f06087e1942214bf4e0b2ab05508a0ceb1b567 at: https://66ed21fe1e173208419880a3--layer5.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 68f6279b76d141fc5b75f4af4633e483d0dd7654 at: https://66ed46382004a334342e8804--layer5.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 178be1cde78f5a698c6cb64ee9ae34a1eec8adf5 at: https://66ed7becca1d517c866ea778--layer5.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit eb1c203c6d79cfd3eee1d6fc96db039beec426b9 at: https://66f41bccc5216704b396f5d0--layer5.netlify.app

File: src/sections/Home/Banner-4/index.js [hunk 4]
```diff
@@ -67,21 +67,21 @@ const Banner1 = (props) => {
...
                 </VintageBox>
               </span>
             </Container>
           </Col>
           {hasMounted && window.innerWidth > 760 && (
-            <Col sm={4} lg={6} className="section-title-wrapper video-col">
+            <Col $sm={4} $lg={6} className="section-title-wrapper video-col">
               <ReactPlayer
-                url="https://youtu.be/034nVaQUyME?si=Yya8m6i7JUoSdZm4"
+                $url="https://youtu.be/034nVaQUyME?si=Yya8m6i7JUoSdZm4"
```
Comment (Ashparshp) [inline src/sections/Home/Banner-4/index.js:84]: Video..

**Comment by l5io** [comment]
🚀 Preview for commit 381b9baeebd4249be113f9b844c257c20b3c71a1 at: https://66f541d2c1584cce15f4d5f2--layer5.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit b6544b2f2c51563226128413bb5c8eb85440ec9b at: https://66f56ee3a9fae7132bfeafcd--layer5.netlify.app

**Pull request closed without merging**